### ISSUE TITLE
Unclosed socket on macOS

### ISSUE THREAD

**Issue by amelchio** [GadgetReactor/pyHS100#65]
I get many warnings like the following when I turn on debugging on macOS:

```
/Users/homeassistant/.homeassistant/deps/pyHS100/pyHS100.py:71: ResourceWarning: unclosed <socket.socket fd=14, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('10.0.1.5', 55327), raddr=('10.0.1.13', 9999)>
  request={target: {cmd: arg}}
```

They disappear if I change the shutdown like suggested [in this comment](https://github.com/GadgetReactor/pyHS100/pull/23#issuecomment-271905451), i.e. `close` even if the `shutdown` fails.

**Comment by rytilahti** [comment]
Which version are you running on? This should have been fixed already with #41 I suppose.

**Comment by amelchio** [comment]
The change in #41 does not call `close` if `shutdown` fails.

**Comment by rytilahti** [comment]
Ok, pushed a fix in #88.

**Comment by amelchio** [comment]
Thank you!

**Comment by rytilahti** [comment]
Not a problem, my mistake by overlooking the issue.

**Comment by rytilahti** [comment]
This is closed by 79f3b44 which will be part of the upcoming release, thanks for your input!
